protocols/gossipsub/regen_structs_proto.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+../../scripts/protobuf/gen.sh src/rpc.proto
```

Comment:
I think we should switch to `prost`, like the rest of rust-libp2p.

Reply:
Agree, done.